lib/algos/list/Extension.js

```diff
@@ -30,22 +47,28 @@ class Extension {
      *
      * @param {Object} value - The JSON value of a listing item
      *
-     * @return {Object} The value without the filtered-out fields.
+     * @return {Object} The value that may have been trimmed of some
+     * heavy unused fields, or left untouched (depending on size
+     * heuristics)
      */
-    filterContent(value) {
+    trimMetadata(value) {
         let ret = undefined;
-        try {
-            ret = JSON.parse(value);
-            delete ret.location;
-            ret = JSON.stringify(ret);
-        } catch (e) {
-            this.logger.warn('Could not parse Object Metadata while listing',
-                             { err: e.toString() });
+        if (value.length >= TRIM_METADATA_MIN_BLOB_SIZE) {
+            try {
+                ret = JSON.parse(value);
+                delete ret.location;
+                ret = JSON.stringify(ret);
+            } catch (e) {
+                // Prefer returning an unfiltered data rather than
+                // stopping the service in case of parsing failure.
+                // The risk of this approach is a potential
+                // reproduction of MD-692, where too much memory is
```

Comment:
Might need rewording into S3C-1985, though I think the link MD-692 should still be valid.